snap-gpf/src/main/java/org/esa/snap/core/gpf/Operator.java

```diff
@@ -96,6 +96,15 @@ protected Operator() {
         context = new OperatorContext(this);
     }
 
+    /**
```

Comment:
Actually, this was intentionally not public. Everything need should be exposed by the Operator class.

Reply:
OK I workaround with java reflectivity to access at Operator method.
Is there a risk with java reflectivity on the side of snap engine?

Reply:
Why do you need to access this? It wasn't necessary in the last 15 years. So, I wonder why it is necessary now.

Reply:
Hi Marco,
Yesterday, I removed only the getContext() in Operator class. I did and will not modify the Operator class.
I have used java reflectivity to have access to the Operator class as you can see in the link below:
https://github.com/senbox-org/snap-graph-editor-extension/blob/main/snap-graph-editor/src/main/java/org/esa/snap/grapheditor/ui/components/NodeGui.java

For example:
Operator operator ...
operator.getClass().getMethod("getTargetProduct").invoke(operator)

I don't know if this could have effect on Operator class as I do not see any other solution to run the new graph builder because the program was initiated by Martino.

Best regards,
Florian Douziech

Reply:
Actually, it should not be necessary to use reflection. But if it works - Go ahead.
I'll try to have a look at the code soon. I might see another way to solve it.
In general, GPF.createProduct(...) should be used.
Maybe here something special is necessary.

Reply:
As I thought, you should be able to use GPF.createProduct(...)
The parameters can be stored in a map and then passed to this method. The same for the source products.
Have a look at the class DefaultSingleTargetProductDialog or even better the ReprojectionDialog in SNAP Desktop. Check the createTargetProduct method. maybe the OperatorParameterSupport caall can also be of help for you.